src/Exceptions/DefectiveConfiguration.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Spatie\Image\Exceptions;
+
+use Exception;
+
+class DefectiveConfiguration extends Exception
+{
+    public static function invalidTemporaryDirectory($dirPath)
+    {
+        if (!is_dir($dirPath)) {
+            $reason = "is not a directory";
+        } elseif (!is_writable($dirPath)) {
```

Comment:
In general I don't like `else` statements`. Extract this to a new method `getReason` where you can work with early returns.